src/main/kotlin/io/zerobase/smarttracing/resources/EmailSender.kt

```diff
@@ -0,0 +1,5 @@
+package io.zerobase.smarttracing.resources
+
+interface EmailSender {
+    fun sendEmail(arr: ByteArray)
```

Comment:
this interface is too coupled to exactly how SES works. the point of an abstraction is to abstract things. this should probably be

```kotlin
fun sendEmail(toAddress: String body: EmailTemplate, attachment = Array<byte>? = null)
```

where `EmailTemplate` is something like

```kotlin
sealed class EmailTemplate {
    abstract fun byte[] render()
}
...
```

we could also do something like 
```kotlin
fun sendEmail(toAddress: String, body: String, contentType: String = "text/html", attachment: Array<Byte>? = null)
```

Reply:
we will have to talk about this because i am confused here, this interface is only in case we want to change the underlying technology used to send the email after it is already packed into the binary state, the email class packs the data into the binary state and sends it to the email sender

Reply:
no, the interface is *not* to change tech for sending after it's already packed into a binary state. the whole point is to be able to change out email providers. if we switch to something like mandril, we won't be sending them a packaged binary stream, we will send them an api request that contains various fields. this interface makes a huge (and wrong) assumption about how an email provider would be accepting data from us.

Reply:
ok than in this case should emailsender not be on the email, and have amazonSES code inside the email sender?